File: src/helper/apis/chatroom.js
/**
 * 聊天室 API —— 家庭组成员互聊
 *
 * 与 chat.js（AI 对话）完全独立。
<<<<<<< HEAD
 */
import $ajax from '../ajax'
import config from './config'
import userIdentity from '../userIdentity'

const baseUrl = config.baseUrl

function uid() {
  return userIdentity.getUserId() || '1'
}

=======
 * user_id 由 ajax.js 自动通过 X-User-Id Header 注入，无需手动传递。
 */
import $ajax from '../ajax'
import config from './config'

const baseUrl = config.baseUrl

>>>>>>> si
export default {
  /**
   * 获取聊天列表（私聊 + 群聊）
   * @returns {Promise<{conversations: Array}>}
   */
  getConversations() {
<<<<<<< HEAD
    return $ajax.get(`${baseUrl}/chatroom/conversations`, {
      user_id: uid(),
    })
=======
    return $ajax.get(`${baseUrl}/chatroom/conversations`)
>>>>>>> si
  },

  /**
   * 发送消息
   * @param {object} params
   * @param {number|null} params.conversation_id - 已有会话 ID
   * @param {string|null} params.receiver_id - 私聊接收者（新建时必传）
   * @param {string} params.chat_type - "private" | "group"
   * @param {string} params.content - 消息内容
   * @returns {Promise<{conversation_id, message_id, content, created_at}>}
   */
  sendMessage({ conversation_id, receiver_id, chat_type, content }) {
    return $ajax.post(`${baseUrl}/chatroom/send`, {
<<<<<<< HEAD
      user_id: uid(),
=======
>>>>>>> si
      conversation_id: conversation_id || null,
      receiver_id: receiver_id || null,
      chat_type,
      content,
    })
  },

  /**
   * 拉取消息（支持增量轮询）
   * @param {number} conversationId - 会话 ID
   * @param {number} afterId - 只拉取 id > afterId 的消息，0=首次加载
   * @returns {Promise<{conversation_id, messages: Array, has_more: boolean}>}
   */
  getMessages(conversationId, afterId = 0) {
    return $ajax.get(`${baseUrl}/chatroom/messages`, {
      conversation_id: conversationId,
<<<<<<< HEAD
      user_id: uid(),
=======
>>>>>>> si
      after_id: afterId,
      limit: 50,
    })
  },
}

File: backend/app/schemas/chatroom.py
"""
聊天室（家庭组成员互聊）的 Pydantic 模型

与 /chat（AI 对话）完全解耦，共用 conversations + chat_messages 表，
通过 Conversation.type 字段区分。
"""
from datetime import datetime
from pydantic import BaseModel, Field


# ── 发送消息 ──

class SendMessageRequest(BaseModel):
    """发送消息请求（user_id 由 X-User-Id Header 传递，不在此模型中）"""
    conversation_id: int | None = Field(None, description="已有会话 ID，不传则新建")
    receiver_id: str | None = Field(None, description="私聊接收者 user_id（新建私聊时必传）")
    chat_type: str = Field(default="group", description="会话类型: private / group")
    content: str = Field(..., min_length=1, max_length=2000, description="消息内容")


class SendMessageResponse(BaseModel):
    """发送消息响应"""
    conversation_id: int
    message_id: int
    content: str
    created_at: datetime | None = None


# ── 拉取消息（轮询） ──

class MessageOut(BaseModel):
    """单条消息"""
    id: int
    sender_id: str  # 发送者 user_id
    sender_nickname: str = ""
    content: str
    created_at: datetime | None = None


class MessagesResponse(BaseModel):
    """消息列表响应"""
    conversation_id: int
    messages: list[MessageOut]
    has_more: bool = False


# ── 聊天列表 ──

class ConversationItem(BaseModel):
    """聊天列表项"""
    id: int
    type: str  # private / group
    title: str | None = None
    last_message: str | None = None
    last_time: datetime | None = None
    # 私聊专用
    peer_user_id: str | None = None
    peer_nickname: str | None = None
    # 群聊专用
    member_count: int = 0


class ConversationListResponse(BaseModel):
    """聊天列表响应"""
    conversations: list[ConversationItem]
